src/javax/microedition/rms/RecordStore.java

```diff
@@ -548,6 +548,7 @@ public int nextRecordId() throws InvalidRecordIDException
 			//System.out.println("> Next Record ID (idx:"+index+" cnt:"+count+")");
 			if(keepupdated) { rebuild(); }
 			if(index>=count) { throw(new InvalidRecordIDException()); }
+			if(index < 0) return elements[0]; // If an invalid index is received, truncate it to 0.
```

Comment:
That's not what 'truncate' means.  Also, how would `index` ever become less than 0?

Reply:
Yeah, i agree that using "truncate" there doesn't make sense.

And as for the index becoming less than 0, a game i tested some days ago (Pinball 3D Timeshock) does taht:
![image](https://user-images.githubusercontent.com/52329050/173636020-b19268ca-fcd2-4dd1-bdcf-2ad1f653a962.png)



It's the game that i fixed by adding that index check. I have no idea why it tries to access a recordId on index -1, but forcing it to move to index 0 allowed it to save its data afterwards:
![image](https://user-images.githubusercontent.com/52329050/173636345-921f8195-0d91-4cc0-8444-fedf8f579411.png)
![image](https://user-images.githubusercontent.com/52329050/173636400-196af37c-df77-4f3c-ba5c-b6ab9d8d8e42.png)

Previously it would only save the "highScore" data, not the "CameraType".

Still, that might not be the best solution to cases like those, i'm open to suggestions.

Reply:
It looks like there are a few places in `enumeration` where it could potentially go below zero.  `previousRecord` and `rebuild` both assume count is greater than zero.  I'm surprised this escaped notice until now.  Count should also be initialized in the constructor...  ugh...  I'll see about updating this later today.

Reply:
I can do those if you want, should help keeping things cleaner as they'll all be in the same PR.

Reply:
Don't worry about the number of pull requests.  I'd rather see a bunch of smaller ones than a big omnibus.  If something needs reverted, it's helpful if the PRs are more focused.  Ideally, a PR will address a specific issue.  It's not uncommon for more experienced  contributors to create an issue right before they submit a PR that addresses it.  

It also makes the review process simpler.  It takes a lot of time to go over these, and I know I'm not spending nearly enough time on most of the files just because of how many there are.  This one alone has something like 22 files changed.  I'd bet that most of these files would have already been approved had they been standalone or in smaller groups. 

Remember that most of this project is tending toward an ideal.  Just like an emulator has the hardware as a standard, this project has the standards as a standard.  The goal isn't to get a game to work, but to get closer to that ideal.   What matters is correctness, above all.  It wasn't uncommon early on to have correct changes break games.  

Don't try to make the game work. That's the easy thing to do, but it's usually wrong.  It's better to find out what problems with our implementation are keeping the game from working.

Reply:
Yeah i'm not worried about making things work with that change above. I just tried to do a few patches to the RecordStore file based on your pointers and it seems to have worked correctly.

And i'll definitely keep that smaller PR advice in mind.